storage/cloud-client/snippets.py

```diff
@@ -0,0 +1,300 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to perform basic operations on blobs
+(objects) in a Google Cloud Storage bucket.
+
+For more information, see the README.md under /storage  and the documentation
+at https://cloud.google.com/storage/docs.
+"""
+
+import argparse
+import datetime
+
+from gcloud import storage
+
+
+def create_bucket(bucket_name):
+    """Creates a new bucket."""
+    storage_client = storage.Client()
+    bucket = storage_client.create_bucket(bucket_name)
+    print('Bucket {} created'.format(bucket.name))
+
+
+def delete_bucket(bucket_name):
+    """Deletes a bucket. The bucket must be empty."""
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    bucket.delete()
+    print('Bucket {} deleted'.format(bucket.name))
+
+
+def list_blobs(bucket_name):
+    """Lists all the blobs in the bucket."""
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+
+    blobs = bucket.list_blobs()
+
+    for blob in blobs:
+        print(blob.name)
+
+
+def list_blobs_with_prefix(bucket_name, prefix, delimiter=None):
+    """Lists all the blobs in the bucket that begin with the prefix.
+
+    This can be used to list all blobs in a "folder", e.g. "public/".
+
+    The delimiter argument can be used to restrict the results to only the
+    "files" in the given "folder". Without the delimiter, the entire tree under
+    the prefix is returned. For example, given these blobs:
+
+        /a/1.txt
+        /a/b/2.txt
+
+    If you just specify prefix = '/a', you'll get back:
+
+        /a/1.txt
+        /a/b/2.txt
+
+    However, if you specify prefix='/a' and delimiter='/', you'll get back:
+
+        /a/1.txt
+
+    """
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+
+    blobs = bucket.list_blobs(prefix=prefix, delimiter=delimiter)
+
+    print('Blobs:')
+    for blob in blobs:
+        print(blob.name)
```

Comment:
mentioned this before, but the fact that every instance of print(blob) seems to be print(blob.name) makes me think the __str__ method of blog should should return name